src/fluxbox.cc

```diff
@@ -1120,39 +1120,7 @@ void Fluxbox::save_rc() {
     } else
         cerr<<_FB_CONSOLETEXT(Fluxbox, BadRCFile, "rc filename is invalid!", "Bad settings file")<<endl;
 
-
-    ScreenList::iterator it = m_screens.begin();
-    ScreenList::iterator it_end = m_screens.end();
-    for (; it != it_end; ++it) {
-        BScreen *screen = *it;
-
-        std::string workspaces_string("session.screen");
-        workspaces_string += FbTk::StringUtil::number2String(screen->screenNumber());
-        workspaces_string += ".workspaceNames: ";
-
-        // these are static, but may not be saved in the users resource file,
-        // writing these resources will allow the user to edit them at a later
-        // time... but loading the defaults before saving allows us to rewrite the
-        // users changes...
-
-        const BScreen::WorkspaceNames& names = screen->getWorkspaceNames();
-        for (size_t i=0; i < names.size(); i++) {
-            workspaces_string += FbTk::FbStringUtil::FbStrToLocale(names[i]);
-            workspaces_string += ',';
-        }
-
-        XrmPutLineResource(&new_rc, workspaces_string.c_str());
-
-    }
-
-    XrmDatabase old_rc = XrmGetFileDatabase(dbfile.c_str());
-
-    XrmMergeDatabases(new_rc, &old_rc);
```

Comment:
idk why XRM was used itfp - it seems inherited from the blackbox code and my guess is that it's because of interoperability?
> _NET_DESKTOP_NAMES
> _NET_DESKTOP_NAMES, UTF8_STRING[]
> The names of all virtual desktops. This is a list of NULL-terminated strings in UTF-8 encoding [[UTF8](https://specifications.freedesktop.org/wm-spec/wm-spec-1.4.html#UTF8)]. This property MAY be changed by a Pager or the Window Manager at any time.

So maybe the idea was that whever sets them first relies on a common database.
What should happen instead is that the property is first test and, if present, inherited into the local data - ie. the alignment happens along the netwm spec.
Then FB can save it in its own resources.

Reply:
@luebking yah, kind of: xrm is working code which can store / read data from a config file. but over time fluxbox has accumulated multiple different config files with different syntax flavours and lots of code is responsible for reading / writing those files.